overlord/snapstate/backend/snapdata.go

```diff
@@ -103,19 +113,70 @@ func (b Backend) untrashData(snap *snap.Info, opts *dirs.SnapDirOptions) error {
 	return err
 }
 
+// removeDirs tries to remove all directories even if there is an error.
+//
+// Note: If there is an error only the first one is returned.
 func removeDirs(dirs []string) error {
+	var firstErr error
 	for _, dir := range dirs {
-		if err := os.RemoveAll(dir); err != nil {
-			return err
+		if err := os.RemoveAll(dir); err != nil && firstErr == nil {
+			firstErr = err
 		}
 	}
 
-	return nil
+	return firstErr
+}
+
+// baseSystemDataHomeDirs returns the per-user base directory for snap data under system.homedirs.
+//
+// More info: https://snapcraft.io/docs/home-outside-home.
+func baseSystemDataHomeDirs(name string, opts *dirs.SnapDirOptions) ([]string, error) {
+	var found []string
+	ssp, err := sysparams.Open("")
```

Comment:
This should probably be `sysparms.Open(dirs.GlobalRootDir)`, no?

Reply:
we have code like this also in sandbox/apparmor, but I'm also a bit confused because the intention was for the file to be used by processes that are not snapd, not for snapd itself to consume the file as the same info should be in the system option, but it seems we do this already :(

Reply:
anyway as I discussed with @ZeyadYasser I asked him to deal with the alternative homedirs in a different PR because we probably have more places that need fixing with matching spread test than just the removal code

Reply:
Thanks, that makes sense and I support this (in fact I was thinking about leaving the same comment but then got interrupted for a meeting :)

Reply:
Thamks, I reverted the changes and will address `system.homedirs` in another PR.